plugins/pgpointcloud/io/PgWriter.cpp

```diff
@@ -230,8 +230,10 @@ uint32_t PgWriter::SetupSchema(uint32_t srid)
     // Do any of the existing schemas match the one we want to use?
     if (schema_count > 0)
     {
-        PGresult *result = pg_query_result(m_session,
-            "SELECT pcid, schema FROM pointcloud_formats");
+        oss.str("");
+        oss.clear();
+        oss << "SELECT pcid, schema FROM pointcloud_formats WHERE srid = " << srid;
+        PGresult *result = pg_query_result(m_session, oss.str());
```

Comment:
Here I'm using oss to construct the query. oss.str("") and oss.clear() ensure that the stream is really empty.
The modified SQL query only selects those schema entries, that have the wanted SRS column. This modification will allow a new row to be generated, when the same schema is used with a different SRS. (Generation of the new entry is performed below with the INSERT statement (line 288ff).